docs/troubleshooting_guide.md

```diff
@@ -0,0 +1,358 @@
+        RX errors 0  dropped 0  overruns 0  frame 0
+        TX packets 5670  bytes 238460 (232.8 KiB)
+        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0
+```
+
+#### 4. ASM candidate disks having a header containing old DG data
+
+* Error:
+```bash
+SEVERE:  [Jan 20, 2021 7:42:09 AM] [FATAL] [INS-30530] Following specified disks have invalid header status: 
+[/dev/asmdisks/DATA_7539397476, /dev/asmdisks/DATA_7539397477, /dev/asmdisks/DATA_7539397478, /dev/asmdisks/DATA_7539397531]
+```
+
+* Fix: Possibly the last time cleanup-oracle.sh was run, it didn't go thru' till the end, as it zeroes out headers cleanly for you.
+
+So, manually following can be run to cleanup as root:
+```bash
+for i in /dev/asmdisks/DATA_7539397476, /dev/asmdisks/DATA_7539397477, /dev/asmdisks/DATA_7539397478, /dev/asmdisks/DATA_7539397531 ; \
+do echo "checking disk $i";dd if=$i bs=128 count=1 | od -a ;echo; \
+done
+```
+
+You will see something like this which shows that the disk headers are not empty and it has remnant DG information:
+```bash
+checking disk /dev/asmdisks/DATA_7539397476
+1+0 records in
+1+0 records out
+128 bytes (128 B) copied, 0.000275708 s, 464 kB/s
+0000000 soh stx soh soh nul nul nul nul nul nul nul nul   L   .  us can
+0000020 nul nul nul nul nul nul nul nul nul nul nul nul nul nul nul nul
+0000040   O   R   C   L   D   I   S   K nul nul nul nul nul nul nul nul
+0000060 nul nul nul nul nul nul nul nul nul nul nul nul nul nul nul nul
+0000100 nul nul etx dc3 nul nul soh etx   D   A   T   A   _   0   0   0
+0000120   0 nul nul nul nul nul nul nul nul nul nul nul nul nul nul nul
+0000140 nul nul nul nul nul nul nul nul   D   A   T   A nul nul nul nul
+0000160 nul nul nul nul nul nul nul nul nul nul nul nul nul nul nul nul
+0000200
+```
+
+* Cleanup as:
+```bash
+#First backup:
+dd if=/dev/asmdisks/DATA_7539397476 of=dev_asmdisks_DATA_7539397476.txt bs=1048576 count=100
+
+#Then cleanup:
+for i in /dev/asmdisks/DATA_7539397476 /dev/asmdisks/DATA_7539397477 /dev/asmdisks/DATA_7539397478 /dev/asmdisks/DATA_7539397531; \
+do dd if=/dev/zero of=$i bs=1048576 count=100 ;echo Done blowing up header for disk $i; done
+```
+
+#### 5. Existing ORACLE_HOME
+
+* Error:
+```bash
+"Launching Oracle Grid Infrastructure Setup Wizard...", "", "[WARNING] [INS-40109] The specified Oracle Base location is not empty on this server.", "   ACTION: Specify an empty location for Oracle Base.", "[WARNING] [INS-32047] The location (/u01/app/oraInventory) specified for the central inventory is not empty.", "   ACTION: It is recommended to provide an empty location for the inventory.", "[FATAL] [INS-13019] Some mandatory prerequisites are not met. These prerequisites cannot be ignored."
+```
+
+* Fix:
+Rerun `cleanup-oracle.sh` and ensure it finished successfully (it can fail at the part where it tries to umount the u01 mountpoint and exits at that point)
+
+#### 6. First time connection errors to the BM from cloud VM
```

Comment:
Can this be made less BM-specific?  Make sure you have a user and appropriate authentication set up on the target host.  By default it uses your own Unix user but can use `--instance-ssh-user` to use a different-user, and `--instance-ssh-key` to specify a SSH private key file to use.

Reply:
Done